test/unit/BuilderTest.php

```diff
@@ -564,15 +564,61 @@ public function withClaimMustKeepAFluentInterface()
     /**
      * @test
      *
+     * @param string $name
+     * @param mixed $value
+     * @param mixed $expected
+     *
      * @covers ::__construct
      * @covers ::withClaim
      * @covers \Lcobucci\JWT\Token\RegisteredClaimGiven
+     *
+     * @dataProvider dataWithClaimDeprecationNotice
      */
-    public function withClaimShouldThrowExceptionWhenTryingToConfigureARegisteredClaim()
+    public function withClaimShouldSendDeprecationNoticeWhenTryingToConfigureARegisteredClaim($name, $value, $expected)
+    {
+        $key = $this->createMock(Key::class);
+        $signature = $this->createMock(Signature::class);
+        $signature
+            ->expects(static::once())
+            ->method('hash')
+            ->willReturn('--hash--')
+        ;
+
+        $signer = $this->createMock(Signer::class);
+        $signer
+            ->expects(static::once())
+            ->method('sign')
+            ->willReturn($signature)
+        ;
+
+
+
+        $this->expectDeprecation('The use of the method "withClaim" is deprecated for registered claims. Please use dedicated method instead.');
+
+        $token = $this
+            ->createBuilder()
+            ->withClaim($name, $value)
+            ->getToken($signer, $key)
+        ;
+
+        self::assertEquals($expected, $token->claims()->get($name));
```

Comment:
Please use the public API instead of mocking here (create the token using `getToken()` as done in other tests).

Reply:
Oh OK! I mised that method :+1: